emu/src/cpu/arm7tdmi.rs

```diff
@@ -680,6 +678,48 @@ impl Arm7tdmi {
 
         Some(SIZE_OF_THUMB_INSTRUCTION)
     }
+
+    pub(crate) fn hi_reg_operation_branch_ex(&mut self, op_code: ThumbModeOpcode) -> Option<u32> {
+        let op = op_code.get_bits(8..=9);
+        let h1 = op_code.get_bit(7);
+        let h2 = op_code.get_bit(6);
+        let rs_hs = op_code.get_bits(3..=5);
+        let _rd_hd = op_code.get_bits(0..=2);
+
```

Comment:
We could do something like:

```rust
let destination_register: usize = (h1 as u16 * 8 + rd_hd).try_into().unwrap();
let source_register: usize = (h2 as u16 * 8 + rs_hs).try_into().unwrap();
```

So we avoid the ifs later

Reply:
Nice idea!